Preprocessing/BOT_MAKE_TRANSCRIPTION.py

```diff
@@ -1,42 +1,89 @@
 import argparse
-import os
+import logging
+from pathlib import Path
 import subprocess
 
 
-def przetworz_folder(folder_wejsciowy: str, folder_wyjsciowy: str) -> None:
-    model = "large-v3"
-    jezyk = "Polish"
-    urzadzenie = "cuda"
+class AudioProcessor:
+    DEFAULT_MODEL = "large-v3"
+    DEFAULT_LANGUAGE = "Polish"
+    DEFAULT_DEVICE = "cuda"
+    SUPPORTED_EXTENSIONS = {".wav", ".mp3"}
 
-    for sciezka_katalogu, _, lista_plikow in os.walk(folder_wejsciowy):
-        for plik in lista_plikow:
-            if plik.endswith((".wav", ".mp3")):
-                sciezka_wejsciowa = os.path.join(sciezka_katalogu, plik)
-                nazwa_pliku_wyjsciowego = plik.split(".")[0] + "_przetworzone.wav"
-                sciezka_wyjsciowa = os.path.join(
-                    folder_wyjsciowy, os.path.relpath(sciezka_katalogu, folder_wejsciowy),
-                    nazwa_pliku_wyjsciowego,
-                )
+    def __init__(self, input_folder: str, output_folder: str, model: str = None, language: str = None, device: str = None):
+        self.input_folder = Path(input_folder)
+        self.output_folder = Path(output_folder)
+        self.model = model or self.DEFAULT_MODEL
+        self.language = language or self.DEFAULT_LANGUAGE
+        self.device = device or self.DEFAULT_DEVICE
+        self.logger = self.setup_logger()
```

Comment:
- Jak coś jest optional to Optional[str]
- To nie musi być Optional, bo przecież domyślne argi miały być w argparserze, więc po co komplikować sobie życie, klasa ma być porpawnie zainicjowana i tyle